From 501432d7897b7151d9a2222b5258e30a8abdd51f Mon Sep 17 00:00:00 2001
From: Ziyuan Xu <xzy.xu@rock-chips.com>
Date: Sat, 7 Nov 2020 17:03:37 +0800
Subject: [PATCH] ARM: dts: rv1126: set isp_tb reserver memory close to
 ramdisk_c

Test on DDR 512MB

[    0.233910] OF: fdt: Reserved memory: reserved region for node 'trust@0': base 0x00000000, size 2048 KiB
[    0.233914] fdt_reserved_mem_save_node trust@0, 0-200000
[    0.233926] OF: fdt: Reserved memory: reserved region for node 'trust@200000': base 0x00200000, size 32 KiB
[    0.233930] fdt_reserved_mem_save_node trust@200000, 200000-8000
[    0.233948] OF: fdt: Reserved memory: reserved region for node 'ramoops@210000': base 0x00210000, size 960 KiB
[    0.233952] fdt_reserved_mem_save_node ramoops@210000, 210000-f0000
[    0.233966] OF: fdt: Reserved memory: reserved region for node 'rtos@300000': base 0x00300000, size 1024 KiB
[    0.233969] fdt_reserved_mem_save_node rtos@300000, 300000-100000
[    0.233982] OF: fdt: Reserved memory: reserved region for node 'ramdisk@2800000': base 0x02800000, size 49152 KiB
[    0.233986] fdt_reserved_mem_save_node ramdisk@2800000, 2800000-3000000
[    0.234000] OF: fdt: Reserved memory: reserved region for node 'ramdisk@5800000': base 0x05800000, size 20480 KiB
[    0.234003] fdt_reserved_mem_save_node ramdisk@5800000, 5800000-1400000
[    0.234017] OF: fdt: Reserved memory: reserved region for node 'rkisp@08000000': base 0x10000000, size 131072 KiB
[    0.234020] fdt_reserved_mem_save_node rkisp@08000000, 10000000-8000000
[    0.234033] OF: fdt: Reserved memory: reserved region for node 'mmc@20f000': base 0x0020f000, size 4 KiB
[    0.234036] fdt_reserved_mem_save_node mmc@20f000, 20f000-1000
[    0.234048] OF: fdt: Reserved memory: reserved region for node 'mmc@500000': base 0x00500000, size 1024 KiB
[    0.234052] fdt_reserved_mem_save_node mmc@500000, 500000-100000

For 4K camera, we should reserved 172MB CMA buffer for ISP. In this scenario,
the system can't find a free area that it's enough for 172MB.

Signed-off-by: Ziyuan Xu <xzy.xu@rock-chips.com>
Change-Id: Ifc9ed49a337c91474842ab3a47783509d754e0d8
---
 arch/arm/boot/dts/rv1126-bat-ipc-v10.dts   | 2 +-
 arch/arm/boot/dts/rv1126-thunder-boot.dtsi | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/rv1126-bat-ipc-v10.dts b/arch/arm/boot/dts/rv1126-bat-ipc-v10.dts
index c27f9b292425..bd731c47a4cc 100644
--- a/arch/arm/boot/dts/rv1126-bat-ipc-v10.dts
+++ b/arch/arm/boot/dts/rv1126-bat-ipc-v10.dts
@@ -351,7 +351,7 @@
 };
 
 &rkisp_thunderboot {
-	reg = <0x10000000 (36 * 0x00100000)>;
+	reg = <0x08000000 (128 * 0x00100000)>;
 };
 
 &rkisp_vir0 {
diff --git a/arch/arm/boot/dts/rv1126-thunder-boot.dtsi b/arch/arm/boot/dts/rv1126-thunder-boot.dtsi
index 01566d31ef22..7eb994b2352b 100644
--- a/arch/arm/boot/dts/rv1126-thunder-boot.dtsi
+++ b/arch/arm/boot/dts/rv1126-thunder-boot.dtsi
@@ -46,8 +46,8 @@
 			reg = <0x05800000 (20 * 0x00100000)>;
 		};
 
-		rkisp_thunderboot: rkisp@10000000 {
-			reg = <0x10000000 (144 * 0x00100000)>;
+		rkisp_thunderboot: rkisp@08000000 {
+			reg = <0x08000000 (144 * 0x00100000)>;
 		};
 	};
 
-- 
2.35.3

